src/os/bluestore/BitAllocator.h

```diff
@@ -530,10 +530,10 @@ class BitAllocator:public BitMapAreaIN{
   bool is_allocated_dis(ExtentList *blocks, int64_t num_blocks);
 
   int64_t size() {
-    return m_total_blocks - m_extra_blocks;
+    return m_total_blocks;
```

Comment:
this does address the issue. but it changes the semantic of `size()`. for example, in `BitAllocator::free_blocks()`, it is expected that `size()` returns the upper bound of usable blocks, but not the "total" number of them.